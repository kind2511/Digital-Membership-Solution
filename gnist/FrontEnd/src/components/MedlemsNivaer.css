.medlems-nivaer-container {
  background-color: #eaeaea;
  border: 1px solid #ccc;
  border-radius: 8px;
  padding: 20px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  overflow: auto; 
}

.section-title {
  color: #333;
  font-size: 20px;
  margin-bottom: 10px;
}

.level-list {
  max-height: 300px; /* Set a maximum height for the list to enable scrolling */
  overflow-y: auto; /* Enable vertical scrolling */
}

.level-item {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 10px 0;
  border-bottom: 1px solid #ccc; 
}

.level-info {
  margin: 0;
}

.level-buttons {
  display: flex;
}

.level-button {
  flex: 1;
  padding: 8px 16px;
  margin-right: 10px; /* Add space between buttons */
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-size: 0.9em;
}

.level-button.edit-button {
  background-color: #FFA500; 
  color: white;
}

.level-button.delete-button {
  background-color: #FF6347; 
  color: white;
}

.level-button:hover {
  opacity: 0.8;
}
