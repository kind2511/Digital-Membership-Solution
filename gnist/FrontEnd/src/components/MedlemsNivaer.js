import React, { useState, useEffect } from 'react';
import './MedlemsNivaer.css';

function MedlemsNivaer() {
    const [levels, setLevels] = useState([]);
    const [newLevelName, setNewLevelName] = useState('');
    const [newLevelPoints, setNewLevelPoints] = useState('');
    const [showSuccessMessage, setShowSuccessMessage] = useState(false);

    useEffect(() => {
        fetchLevels();
    }, []);

    const fetchLevels = async () => {
        try {
            const response = await fetch('http://127.0.0.1:8000/digital_medlemsordning/get_all_levels/');
            if (!response.ok) {
                throw new Error('Failed to fetch levels');
            }
            const data = await response.json();
            setLevels(data);
        } catch (error) {
            console.error(error);
        }
    };

    const handleEditLevel = (levelID) => {
        //TODO
        console.log(`Editing level with ID: ${levelID}`);
    };

    const handleDeleteLevel = (levelID) => {
        //TODO
        console.log(`Deleting level with ID: ${levelID}`);
    };

    const handleAddLevel = async () => {
        try {
            const response = await fetch('http://127.0.0.1:8000/digital_medlemsordning/create_level/', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    name: newLevelName,
                    points: newLevelPoints
                })
            });
            if (!response.ok) {
                throw new Error('Failed to add new level');
            }
            setShowSuccessMessage(true);
            setNewLevelName('');
            setNewLevelPoints('');
            fetchLevels(); // Fetch levels again to update the list
            setTimeout(() => {
                setShowSuccessMessage(false); // Hide the success message after 3 seconds
            }, 3000);
        } catch (error) {
            console.error(error);
        }
    };

    return (
        <div className="medlems-nivaer-container">
            <h2 className="medlems-nivaer-section-title">Medlems Niv책er</h2>
            <div className="section-content">
                <div className="level-list">
                    {levels.map((level, index) => (
                        <div key={index} className="level-item">
                            <p className="level-info">{level.name} - {level.points} Poeng</p>
                            <div>
                                <button className="level-button edit-button" onClick={() => handleEditLevel(level.levelID)}>Redigere</button>
                                <button className="level-button delete-button" onClick={() => handleDeleteLevel(level.levelID)}>Slett</button>
                            </div>
                        </div>
                    ))}
                </div>
                <div className="add-level-form">
                    <input
                        type="text"
                        placeholder="Navn p책 niv책"
                        value={newLevelName}
                        onChange={(e) => setNewLevelName(e.target.value)}
                    />
                    <input
                        type="text"
                        placeholder="Poeng"
                        value={newLevelPoints}
                        onChange={(e) => setNewLevelPoints(e.target.value)}
                    />
                    <button className="level-button add-button" onClick={handleAddLevel}>Legg til nytt niv책</button>
                </div>
            </div>
            {showSuccessMessage && (
                <div className="success-banner">
                    Niva har blitt lagt til
                </div>
            )}
        </div>
    );
}

export default MedlemsNivaer;
