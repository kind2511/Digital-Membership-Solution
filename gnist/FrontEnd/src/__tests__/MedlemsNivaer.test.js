import React from 'react';
import renderer from 'react-test-renderer';
import MedlemsNivaer from '../components/MedlemsNivaer';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// Mock axios for the test
const mock = new MockAdapter(axios);
const levelsMock = [
  { levelID: 1, name: 'Bronze', points: 100 },
  { levelID: 2, name: 'Silver', points: 200 },
  { levelID: 3, name: 'Gold', points: 300 }
];

mock.onGet('http://127.0.0.1:8000/digital_medlemsordning/get_all_levels/').reply(200, levelsMock);
mock.onPost('http://127.0.0.1:8000/digital_medlemsordning/create_level/').reply(201);
mock.onPut('http://127.0.0.1:8000/digital_medlemsordning/edit_level/1/').reply(200);
mock.onDelete('http://127.0.0.1:8000/digital_medlemsordning/delete_level/1/').reply(200);

test('MedlemsNivaer component snapshot', () => {
  const component = renderer.create(<MedlemsNivaer />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
