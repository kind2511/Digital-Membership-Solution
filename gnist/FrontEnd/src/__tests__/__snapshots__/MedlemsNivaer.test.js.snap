// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedlemsNivaer component snapshot 1`] = `
<div
  className="medlems-nivaer-container"
>
  <h2
    className="medlems-nivaer-section-title"
  >
    Medlems Niv책er
  </h2>
  <div
    className="section-content"
  >
    <div
      className="level-list"
    />
    <div
      className="add-level-form"
    >
      <input
        onChange={[Function]}
        placeholder="Navn p책 niv책"
        type="text"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="Poeng"
        type="text"
        value=""
      />
      <button
        className="level-button add-button"
        onClick={[Function]}
      >
        Legg til nytt niv책
      </button>
    </div>
  </div>
</div>
`;
